code/socialDistribution/models/post.py

```diff
@@ -348,6 +348,14 @@ class InboxPost(Post):
     def author_as_json(self):
         """ Gets the author of the post in JSON format. """
         return self._author_json
+    
+    def get_id(self):
+        if type(self.author) is str:
+            authorid = self.author.split('/')[-1]
+        else:
+            authorid = self.author.id
+
+        return f"{API_BASE}/author/{authorid}/posts/{self.id}"
```

Comment:
I'm not sure if I understand what this method does

Reply:
Removed it, got confused. Using public_id now

Reply:
For InboxPosts self.author is a url whereas for LocalPost it is an author object

Reply:
@JawadHossain ya but in the case of a remote author the URL would point to nothing

Reply:
oops removed comment by mistake. thanks guys